### DISCUSSION TITLE
Postal can't parse Persian and Arabic characters

### DISCUSSION THREAD

**Discussion by ahmadrezaei** [postalserver/postal#2601] [Development discussions]
Hi,
when I try to send emails in Persian or Arabic postal can't parse it and when postal send email to http endpoint email body is only question mark.
Can anyone help?

<img width="701" alt="Screenshot 1402-06-15 at 10 22 31" src="https://github.com/postalserver/postal/assets/7928105/d9c1b385-f5d2-48d8-94d1-4f69fa5b1f02">

**Comment by willpower232** [comment]
Can you check in the database and see what the content of the message is? It would be good to know if the database is storing the right characters at all.

Now that I write this, I think someone mentioned something similar recently but I can't see it now.

**Reply by ahmadrezaei** [reply]
Database is storing data right, but GUI can’t parse email content correcly. http endpoint too.

**Reply by AleksY1991** [reply]
I have same issue with German Symbols.
When I click download from UI - works fine.
When I receive either raw or JSON - message is destroyed.

**Reply by willpower232** [reply]
Thanks for checking, it looks like Postal connects to the database using utf8mb4 which should be correct but might not be for you?

It also looks like you can specify `encoding` alongside the rest of the database connection settings to change it.

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/lib/postal/config.rb#L86

I don't really know Ruby on Rails but there might be another encoding setting somewhere else but fiddling with the database one is usually where I start with this kind of problem.

**Comment by AleksY1991** [comment]
Imho there's no need to do anything with the database because when you
click Download - it's normally formatted.
1 - On the website in the admin panel - it's broken
2 - On API - it's broken so it's Backend processing only
[image: image.png]

I have attached the exported file. (Clicked download on the right side)
Which you can check yourself works perfectly

I wasn't able to find contacts from other guys.
Maybe you could ask them?

Best regards,
Aleks Yanovskyi.

**Reply by ahmadrezaei** [reply]
Problem isn't from database, I think postal can't parse raw data correctly.
